velox/exec/OutputBuffer.h

```diff
@@ -74,6 +74,36 @@ class ArbitraryBuffer {
   std::deque<std::shared_ptr<SerializedPage>> pages_;
 };
 
+/// The stats of a DestinationBuffer. The data transferred by the buffer has
+/// two phases:
+/// 1. Buffered: after added via enqueue() and before acked, the data is
+///              buffered in the destination buffer.
+/// 2. Sent: the data is sent and removed from the buffer after it is acked
+///          via acknowledge() or deleteResults().
+struct BufferStats {
+  /// Update stats when a data page is added.
+  void recordAddPage(const SerializedPage& data);
+
+  /// Update stats when a data page is sent (removed from buffer).
+  void recordSendPage(const SerializedPage& data);
+
+  /// If the destination buffer is finished. A finished dest buffer would be
+  /// deleted by the OutputBuffer, so this flag is set by the OutputBuffer.
+  bool finished{false};
+  /// Byte num of data that is currently buffered.
```

Comment:
Separate properties with empty lines for readability.

Comments are redundant as they repeat the names of the variables. The explanation of buffered vs. sent in the struct-level comment is sufficient.

```
int64_t bytesBuffered{0};
int64_t rowsBuffered{0};
int64_t pagesBuffered{0};

int64_t bytesSent{0};
int64_t rowsSent{0};
int64_t pagesSent{0};
```

Reply:
done.